### PR TITLE
Pinned column

### PR DESCRIPTION
**Pull request by etokareva**

### CODE REVIEW

File: docs/column.pin.md [hunk 1]
```diff
@@ -1,12 +1,25 @@
 # qgrid 
 ## markup
 ```html
-<q-grid rows='data'></q-grid>
+<q-grid rows="$ctrl.rows">
+	<q-grid:columns>
+		<q-grid:column isPinned="true" key="name.last" title="Last Name" value="$ctrl.getLastName($row)"></q-grid:column>
```
Comment (vkorolev) [inline docs/column.pin.md]: `is-pinned` in this case :)
But as for me just `pinned` would be enough.

File: docs/column.pin.md [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
 ```
 ## script
 ```javascript
-controller.$inject = ['$scope'];
-function controller($scope){
-	$scope.data = [];
+controller.$inject = ['$scope', 'qgrid'];
+function controller($scope, qgrid){
+  $scope.columns = [
+          {key: 'firstName', label: 'First Name', type: 'text', isVisible: true, isDefault: true, isPinned: true},
```
Comment (vkorolev) [inline docs/column.pin.md]: Consider to use following default values: 
* visible - true
* default - true
* pinned - false

For other cases user will be add configuration

File: docs/column.pin.md (hunk 1 above)
Comment (klumba12) [reply docs/column.pin.md]: yep, we'll make desicion to use or not to use 'is' :)

File: docs/column.pin.md (hunk 2 above)
Comment (klumba12) [reply docs/column.pin.md]: default is reserved word, so maybe use 'is'?

Comment (klumba12) [reply docs/column.pin.md]: it's good practice to use is/can maybe for predicating...

File: docs/column.pin.md (hunk 1 above)
Comment (klumba12) [reply docs/column.pin.md]: maybe pin='left|right' ?

File: docs/column.pin.md (hunk 2 above)
Comment (klumba12) [reply docs/column.pin.md]: pin='left|right' ?

File: docs/column.pin.md (hunk 1 above)
Comment (vkorolev) [reply docs/column.pin.md]: Yeah, i like this in markup

File: docs/column.pin.md (hunk 2 above)
Comment (vkorolev) [reply docs/column.pin.md]: Yes, in JS-configuration we can use is- prefix, it looks good. pin='left|right' looks good too

Comment (etokareva) [reply docs/column.pin.md]: Maybe pin='left|right' for html and isPinned='left|right' for js then?

File: docs/column.pin.md [hunk 3]
```diff
@@ -1,12 +1,25 @@
...
+controller.$inject = ['$scope', 'qgrid'];
+function controller($scope, qgrid){
+  $scope.columns = [
+          {key: 'firstName', label: 'First Name', type: 'text', isVisible: true, isDefault: true, isPinned: 'left'},
+          {key: 'lastName', label: 'Last Name', type: 'text', isVisible: true, isDefault: true, isPinned: 'left'},
+          {key: 'birthDate', label: 'Date of Birth', type: 'date', isVisible: true, isDefault: true},
+          {key: 'location', label: 'Location', type: 'date', isVisible: true, isDefault: true, isPinned: 'right'},
+          {key: 'zipCode', label: 'Zip', type: 'number', isVisible: false, isDefault: false, isPinned: 'right'}
+      ];
+
```
Comment (klumba12) [inline docs/column.pin.md:23]: Ispinned to pin

Comment (klumba12) [reply docs/column.pin.md:23]: Also label to title 😀

Comment (takesanocean) [reply docs/column.pin.md:23]: maybe not `pin="left"` but `pinned="left"`?
Sounds more natural to me.

Comment (klumba12) [reply docs/column.pin.md:23]: pinned it sounds like a boolean state for me :)
however we need to have common style pin vs pinned, canResize vs resizable, canMove vs moveable, isVisible vs visible, isDefault vs 'default' etc.

**Pull request closed without merging**